BrickKit/bricks/src/main/java/com/wayfair/brickkit/viewholder/factory/BrickViewHolderFactory.java

```diff
@@ -0,0 +1,117 @@
+
+import com.wayfair.brickkit.viewholder.BrickViewHolder;
+import com.wayfair.brickkit.brick.BaseBrick;
+import com.wayfair.brickkit.brick.BrickProvider;
+import com.wayfair.brickkit.view.empty.EmptyBrickView;
+import com.wayfair.brickkit.viewholder.EmptyBrickViewHolder;
+
+/**
+ * A factory used to create various types of {@link BrickViewHolder} objects.
+ * <p>
+ * This class takes it's roots from the  Factory Design Pattern.  Instead of using an interface, it
+ * is using the {@link BrickViewHolder} abstract class.  For more information on the pattern,
+ * refer to:
+ * <a href="https://www.tutorialspoint.com/design_pattern/factory_pattern.htm">Factory Design Pattern</a>.
+ * <p>
+ * Copyright © 2017 Wayfair. All rights reserved.
+ */
+public class BrickViewHolderFactory {
+
+    /**
+     * Creates {@link BrickViewHolder} objects based on the {@param #data}.
+     *
+     * @param viewHolderFactoryData - used for instantiation of {@link BrickViewHolder} objects.
+     * @return the created {@link BrickViewHolder}
+     */
+    public BrickViewHolder createBrickViewHolder(
+            @NonNull BrickViewHolderFactoryData viewHolderFactoryData) {
+
+        Context context = viewHolderFactoryData.getParent().getContext();
+        BrickViewHolder viewHolder = null;
+        try {
+            // The following creates the correct view holder, based on the layout res id.  The
+            // strategy is for view types of the default value / zero, an empty brick view holder
+            // is created, providing for it to be returned by the Recycler View adapter instead and
+            // for avoiding app crashes.
+            if (BaseBrick.DEFAULT_LAYOUT_RES_ID >= viewHolderFactoryData.getViewType()) {
+                // By creating and using an empty brick view holder, the view holder can be in
+                // place in scenarios.  i.e. when bricks are null and a view holder cannot be
+                // created.
+                viewHolder = createEmptyBrickViewHolder(context);
+            } else {
+                // Since the view type (layout id) is set, create a standard brick view holder.
+                viewHolder = createViewHolderWithViewType(
+                        viewHolderFactoryData.getParent(),
+                        viewHolderFactoryData.getViewType(),
+                        viewHolderFactoryData.getBrickProvider());
+            }
+        } catch (AssertionError ae) {
+            // This shouldn't happen
+            Log.wtf(viewHolderFactoryData.getLogTag(), "Unable to get the layout inflater. " +
+                    viewHolderFactoryData);
+        } catch (Resources.NotFoundException nfe) {
+            Log.w(viewHolderFactoryData.getLogTag(), "Unable to find the resource. " +
+                    viewHolderFactoryData, nfe);
+        } catch (NullPointerException npe) {
+            Log.w(viewHolderFactoryData.getLogTag(), "The brick is null and shouldn't be." +
+                    viewHolderFactoryData);
```

Comment:
probably should log the npe as well to get the stack trace?

Reply:
Hey @kenyee I updated the log statement to reflect this requested change.  I also updated the log statement for the assertion error to match.